### DISCUSSION TITLE
Best Practice 'Add On Library' >>Create additional 'Extension Library' for Spectre.Console?

### DISCUSSION THREAD

**Discussion by jeffward01** [spectreconsole/spectre.console#784] [Q&A]
Hello all!

I was wondering what is the best practice, or methods to create a Spectre.Console 'add-on' library.  

# Concept
My idea is I would like to create a library called:

`Spectre.Console.FigletFonts`

Essentially the library will be a wrapper around [FigletFonts](https://github.com/xero/figlet-fonts) and allow users to easily add new Figlet fonts. In the same way that Emojis, or Spinners can be 'referenced', these Figlets Fonts will be referenced in the same manner.

# Question
Can I just create my own repo, and push it myself to nuget and Github? or is there a preferred method of doing this?

---

Thanks!

`Note: this can also be added to the 'Idea' topic category`

**Comment by nils-a** [comment]
@jeffward01 there is no best practice currently. 

> Can I just create my own repo, and push it myself to nuget and Github?

Yes.

> ... or is there a preferred method of doing this?

No.

**Comment by phil-scott-78** [comment]
The Spectre.Console name should be reserved for official packages, so we
have been recommending something like "Unofficial.Spectre.Console" for the
name. But truth be told, I'm not sure anyone has gone down this route yet.

**Reply by nils-a** [reply]
As far as I can se, there's some unofficial packages in the `Spectre.Console.Extensions` space, and one particular nucklehead has published two registrar implementations under `Spectre.Console.Registrars`.
